CONVENTIONS.md

```diff
@@ -141,20 +141,129 @@ it is used to manage all elements of the cluster.
-during the standard e2e suite.
-
-The memory request of cluster components should be set at 10% higher than their p90 memory usage over a standard e2e suite execution.
-
-Limits should be not set because components cannot anticipate how they scale in usage in all customer environments and a crashlooping OOM component is something we must detect and handle gracefully regardless. Setting memory limits leaves administrators with no option to react to valid changes usage dynamically.
-
+Kubernetes allows Pods to declare their CPU and memory resource
+requirements in advance using [requests and
+limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/). Requests
+are used to ensure minimum resources are provided and to influence
+scheduling, while limits prevent Pods from consuming resources
+excessively.
+
+Unlike with user workloads, setting limits for cluster components is
+problematic for several reasons:
+
+* Components cannot anticipate how they scale in usage in all customer
+  environments, so setting one-size-fits-all limits is not possible.
+* Setting static limits prevents administrators from responding to
+  changes in their clusters’ needs, such as by resizing control plane
+  nodes to provide more resources.
+* We do not want cluster components to be restarted based on their
+  resource consumption (for example, being killed due to an
+  out-of-memory condition). We need to detect and handle those cases
+  more gracefully, without degrading cluster performance.
+
+Therefore, cluster components *SHOULD NOT* be configured with resource
+limits.
+
+However, cluster components *MUST* declare resource requests for both
+CPU and memory.
+
+Specifying requests without limits allows us to ensure that all
+components receive their required minimum resources and that they are
+able to burst to use more resources as needed. When setting the
+resource requests, we need to balance the size between values that
+ensure the component has the resources it needs to keep running with
+the requirement to be efficient and support small-footprint
+deployments. If the request settings for a component are too low, it
+could be starved for resources in a very busy cluster, resulting in
+degraded performance and lower quality of service. If the request
+settings for a component are too high, it could mean that the
+scheduler cannot find a place to run it, leading to crash looping or
+preventing a cluster from deploying. If the combined resource requests
+for the cluster components are too high, users may not have enough
+resources available to use OpenShift at all, resulting in lost
+business (see especially the [single-node](enhancements/single-node)
+use cases).
+
+Kubernetes resource requests for CPU time are measured in fractional
+"[Kubernetes
+CPUs](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu)"
+expressed using the unit “millicpu”, or 1/1000th of a CPU. For
```

Comment:
Hrm, we almost always call these `millicores`.  I was surprised you used `millicpu`.

Reply:
millicpu came from https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu

It does say that millicore is often used to mean the same thing, but a core and a CPU are only the same when hyperthreading is not enabled. I've updated this section to mention both names.